<?xml version="1.0"?>
<launch>
  <!-- TTC Calculator Node -->
  <node pkg="ttc" type="ttc_calculator" name="ttc_calculator" output="screen">
    <!-- Noise Parameters -->
    <param name="noise_std" type="double" value="0.5" />
    
    <!-- TTC Threshold Parameters -->
    <param name="ttc_warning_threshold" type="double" value="5.0" />
    <param name="ttc_critical_threshold" type="double" value="2.0" />
    
    <!-- Update Rate (Hz) -->
    <param name="update_rate" type="double" value="50.0" />
    
    <!-- Subscriber Topic Names -->
    <param name="static_pose_topic" type="string" value="/vehicle/static/pose" />
    <param name="static_velocity_topic" type="string" value="/vehicle/static/velocity" />
    <param name="static_bbox_topic" type="string" value="/vehicle/static/bounding_box" />
    <param name="manual_pose_topic" type="string" value="/vehicle/manual/pose" />
    <param name="manual_velocity_topic" type="string" value="/vehicle/manual/velocity" />
    <param name="manual_bbox_topic" type="string" value="/vehicle/manual/bounding_box" />
    
    <!-- Publisher Topic Names -->
    <param name="ttc_true_topic" type="string" value="/ttc/true" />
    <param name="ttc_estimated_topic" type="string" value="/ttc/estimated" />
    <param name="ttc_visualization_topic" type="string" value="/ttc/visualization" />
    <param name="ttc_text_topic" type="string" value="/ttc/text" />
    <param name="manual_true_pose_topic" type="string" value="/vehicle/manual/pose_true" />
    <param name="manual_noisy_pose_topic" type="string" value="/vehicle/manual/pose_estimated" />
  </node>
  
  <!-- RViz Visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ttc)/rviz/ttc.rviz"/>
</launch>